test/dynamo/test_higher_order_ops.py

```diff
@@ -544,6 +567,60 @@ def f(x):
         x = torch.randn(3)
         self._test_wrap_simple(f, (x,), 3, expected_opcount=2)
 
+    def test_nested_wrap(self):
+        class MockModule(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.linear = torch.nn.Linear(10, 10)
+
+            def forward(self, x):
+                return self.linear(x)
+
+        mod = MockModule()
+
+        # Two levels of wrap ops
+        def gn(x):
+            return torch.cos(x) + wrap(mod, x)
+
+        def fn(x):
+            return wrap(gn, x)
+
+        self._test_wrap_simple(fn, (torch.randn(10, 10),), 4, expected_opcount=1)
+
+    def test_hooks(self):
+        class ToyModel(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.net = torch.nn.Linear(10, 10)
+
+            def forward(self, x):
+                return self.net(x)
+
+        model = ToyModel()
+        forward_handles = {}
+        activations = dict()
+
+        def save_activations(mod, inp, out):
+            activations[name] = inp
+
+        for name, module in model.named_children():
+            forward_handles[name] = module.register_forward_hook(save_activations)
```

Comment:
Can we add a test for `module.register_full_backward_hook`? I would expect this to cause the entire wrap() to fall back to eager mode (because we traced out the module, there is no way for the full backward hook to actually fire).